sdk/python/kfp/components/structures.py

```diff
@@ -816,14 +816,35 @@ def load_from_component_yaml(cls, component_yaml: str) -> 'ComponentSpec':
             Component spec in the form of V2 ComponentSpec.
         """
 
+        def extract_description(component_yaml: str) -> Union[str, None]:
+            heading = '# Description: '
+            if heading in component_yaml:
+                description = component_yaml.splitlines()[2]
+
```

Comment:
Finding the index would not work using `list.index()`  if we don't know what exactly the string we are indexing is, for example, description might be `#Description : test` or `#Description : testing testing`. I added a `index_of_heading = 2` line to give more context to the value `[2]` since the heading would always be located in the second index

Reply:
Makes sense. Thanks for explaining.